### PR TITLE
test: concurrent users metrics

### PR DESCRIPTION
**Pull request by iadmytro**
# Context

Test for simulation of concurrent wallet restoration workload from specified number of users.

Process
   Test gradually increase load from 1 user to specified number of users
   Time measured till last wallet in the batch restored

Parameters
  MAX_USERS - max number of concurrent wallet restorations in the batch
  RESTORATION_TIMEOUT -  max wait time in milliseconds 
  Groups - (currently hardcoded) users divided by number of tx in wallet to restore

Output written to csv file with 2 columns number of concurrent users and restoration time

### CODE REVIEW

**Review by iccicci — DISMISSED** [review]
LGTM

File: packages/e2e/test/load-test-custom/wallet-restoration/wallet-restoration.test.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    const filteredAddr = result.filter((e: any) => e.tx_count > group[0] && e.tx_count < group[1]);
+    if (filteredAddr < numAddresses) throw new Error('Not enough addresses!');
+    const addresses: GroupedAddress[] = filteredAddr.map(mapToGroupedAddress);
+    const resultsFile = `${resultsFilePrefix}-${group[0]}-${group[1]}-.csv`;
+    for (let users = 1; users <= maxUsers; users++) {
+      const wallets = await initWallets(users, addresses.slice(0, maxUsers));
+      addresses.splice(0, maxUsers);
+      try {
+        await measureRestoration(wallets, resultsFile);
+        wallets.map((w) => w.shutdown());
```
Comment (rhyslbw) [inline packages/e2e/test/load-test-custom/wallet-restoration/wallet-restoration.test.ts:97]:
```suggestion
        wallets.forEach(({ shutdown }) => shutdown());
```

**Review by rhyslbw — CHANGES_REQUESTED** [review]
Couple of lint failures

File: packages/e2e/test/artillery/wallet-restoration/WalletRestoration.ts [hunk 2]
```diff
@@ -64,15 +64,6 @@ const extractAddresses = async (count: number): Promise<AddressesModel[]> => {
   return result;
 };
 
-const mapToGroupedAddress = (addrModel: AddressesModel): GroupedAddress => ({
```
Comment (rhyslbw) [inline packages/e2e/test/artillery/wallet-restoration/WalletRestoration.ts:67]: The extraction of this util should be done first in a separate commit, of type `test`.

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: packages/e2e/test/artillery/wallet-restoration/WalletRestoration.ts (hunk 2 above)
Comment (iadmytro) [reply packages/e2e/test/artillery/wallet-restoration/WalletRestoration.ts:67]: Done

File: packages/e2e/test/load-test-custom/wallet-restoration/wallet-restoration.test.ts (hunk 1 above)
Comment (iadmytro) [reply packages/e2e/test/load-test-custom/wallet-restoration/wallet-restoration.test.ts]: fixed linter errors

File: packages/e2e/test/load-test-custom/wallet-restoration/wallet-restoration.test.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+const fs = require('fs');
+const env = getEnv([...walletVariables, 'DB_SYNC_CONNECTION_STRING']);
+const testLogger: Logger = console;
+const range = (toNum: number) => {
+  // [1 ... toNum]
+  const resArr = [...Array.from({ length: toNum + 1 }).keys()];
+  resArr.shift();
+  return resArr;
+};
+const mapToGroupedAddress = (addrModel: AddressesModel): GroupedAddress => ({
```
[the remark this answers is not in the document]
Comment (iadmytro) [reply packages/e2e/test/load-test-custom/wallet-restoration/wallet-restoration.test.ts]: mapToGroupedAddress moved to util

**Review by mkazlauskas — CHANGES_REQUESTED** [review]
Looks good! Both commits can be scope to single package `test(e2e):`, as there are no changes in other packages

**Comment by iadmytro** [comment]
> Both commits can be scoped to single package `test(e2e)`

Done

**Review by iccicci — APPROVED** [review]
